### PR TITLE
DEPR: DataFrame.resample axis parameter

### LINKED ISSUE #51778: DEPR: axis=1 in DataFrame.window, resample
xref #51203 we've deprecated groupby with axis=1 bc under the hood it does a transpose anyway, so might as well tell users to do that directly.  resample and window do the same thing.  Let's deprecate those too.

### PR DESCRIPTION
**Pull request by dannyi96**
- [X] closes #51778(Replace xxxx with the GitHub issue number)
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [X] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/v2.1.0.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/tests/resample/test_datetime_index.py [hunk 1]
```diff
@@ -641,12 +641,16 @@ def test_resample_dup_index():
...
-    result = df.resample("Q", axis=1).mean()
-    msg = "DataFrame.groupby with axis=1 is deprecated"
-    with tm.assert_produces_warning(FutureWarning, match=msg):
-        expected = df.groupby(lambda x: int((x.month - 1) / 3), axis=1).mean()
-    expected.columns = [Period(year=2000, quarter=i + 1, freq="Q") for i in range(4)]
-    tm.assert_frame_equal(result, expected)
+    warning_msg = "DataFrame.resample with axis=1 is deprecated."
+    with tm.assert_produces_warning(FutureWarning, match=warning_msg):
+        result = df.resample("Q", axis=1).mean()
+        msg = "DataFrame.groupby with axis=1 is deprecated"
```
Comment (mroeschke) [inline pandas/tests/resample/test_datetime_index.py:647]: Could you un-indent the rest of this code?

**Comment by mroeschke** [comment]
From the original issue. The `rolling` axis parameter should be deprecated too

File: pandas/tests/resample/test_datetime_index.py (hunk 1 above)
Comment (dannyi96) [reply pandas/tests/resample/test_datetime_index.py]: sure, done

File: pandas/tests/resample/test_datetime_index.py [hunk 2]
```diff
@@ -729,9 +733,11 @@ def test_resample_axis1(unit):
     rng = date_range("1/1/2000", "2/29/2000").as_unit(unit)
     df = DataFrame(np.random.randn(3, len(rng)), columns=rng, index=["a", "b", "c"])
 
-    result = df.resample("M", axis=1).mean()
-    expected = df.T.resample("M").mean().T
-    tm.assert_frame_equal(result, expected)
+    warning_msg = "DataFrame.resample with axis=1 is deprecated."
+    with tm.assert_produces_warning(FutureWarning, match=warning_msg):
+        result = df.resample("M", axis=1).mean()
+        expected = df.T.resample("M").mean().T
```
[the remark this answers is not in the document]
Comment (dannyi96) [reply pandas/tests/resample/test_datetime_index.py]: sure, done

**Comment by dannyi96** [comment]
> From the original issue. The `rolling` axis parameter should be deprecated too

Yup correct, will work on the `rolling` axis param deprication. 
Was wondering if these two can be taken up in separate PRs ? 
If not, will update this PR with `rolling` changes as well in a few days.

**Comment by dannyi96** [comment]
```
/home/runner/micromamba/envs/test/lib/python3.8/site-packages/numpy/core/include/numpy/npy_1_7_deprecated_api.h:
17:2: warning: #warning "Using deprecated NumPy API, 
disable it with " "#define NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION" [-Wcpp]
build finished with problems, 2 warnings.
```
The docs build seems to have failed. 
Is there any such known issue ? ( seems unrelated to the changes made in this PR )

File: pandas/core/generic.py [hunk 3]
```diff
@@ -11947,11 +12002,29 @@ def ewm(
...
+        axis: Axis | lib.NoDefault = lib.no_default,
         times: np.ndarray | DataFrame | Series | None = None,
         method: str = "single",
     ) -> ExponentialMovingWindow:
-        axis = self._get_axis_number(axis)
+        if axis is not lib.no_default:
+            axis = self._get_axis_number(axis)
+            if axis == 1:
+                warnings.warn(
+                    "DataFrame.ewm with axis=1 is deprecated. Do "
```
[the remark this answers is not in the document]
Comment (mroeschke) [reply pandas/core/generic.py:12076]: You can use `f"{type(self).__name__}.ewm with ..."`

[the remark this answers is not in the document]
Comment (dannyi96) [reply pandas/core/generic.py:12076]: nice ! will update the PR

**Comment by dannyi96** [comment]
seems like this issue has already been worked upon and merged by #51997  ( CC: @jbrockmendel , @mroeschke  )

**Comment by mroeschke** [comment]
Ah yes, sorry @dannyi96. As you mentioned, this PR can still deprecate the axis parameter for resample if you're still interested

**Comment by dannyi96** [comment]
> Ah yes, sorry @dannyi96. As you mentioned, this PR can still deprecate the axis parameter for resample if you're still interested

Sure @mroeschke , have updated the PR

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Nice thanks @dannyi96

**Pull request merged**